daffodil-runtime1-unparser/src/main/scala/org/apache/daffodil/processors/unparsers/SpecifiedLength2.scala

```diff
@@ -460,6 +463,64 @@ class ElementUnusedUnparser(
 
 }
 
+/**
+ * This is basically a simplified version of the ElementUnusedUnparserSuspendableOperation.
+ * For explicit length choices, we do not need to handle length expressions, choiceLength is
+ * simply a non-negative integer.
+ */
+class ChoiceUnusedUnparserSuspendableOperation(
+  override val rd: TermRuntimeData,
+  targetLengthInBits: Long)
+  extends SuspendableOperation {
+
+  /**
+   * determine delta between value length and target length
+   *
+   * and skip that many bits.
+   */
+  override def continuation(ustate: UState) {
+    val e = ustate.currentInfosetNode.asInstanceOf[DIElement]
+    val vl = e.valueLength.lengthInBits.longValue
+    val skipInBits = targetLengthInBits - vl
+    if (skipInBits < 0)
+      UE(ustate, "Data too long by %s bits. Unable to truncate.", -skipInBits)
```

Comment:
I might be worth mention an explicit length choice here in this error message to make it clear that the choice content was too long for the specified length.